hack/.deployUtils

```diff
@@ -327,8 +327,10 @@ configureController() {
     kubectl config use-context kind-${clusterName}
     echo "Initialize local dev setup for the controller on ${clusterName}"
 
-  if [ -n "$MGC_AWS_ACCESS_KEY_ID" ]; then
-    cat <<EOF | kubectl apply -f -
+    if [ "$DNS_PROVIDER" == "aws" ]; then
+        if [ -n "$MGC_AWS_ACCESS_KEY_ID" ]; then
```

Comment:
adding validation here might not be bad thing

```
if [ "$MGC_AWS_ACCESS_KEY_ID" == "" ]; then
     echo "MGC_AWS_ACCESS_KEY_ID is not set"
```
etc

Reply:
Actually, perhaps a small refactor here overall would be better?

like a switch and some functions?
```
case $DNS_PROVIDER in
   aws)
      echo "setting up aws dns provider"
      awsProvider()
      ;;
   gcp)
      echo "setting up gcp dns provider"
      gcpProvider()
      ;;
   *)
     echo "unknown provider"
     exit
     ;;
esac

```

Reply:
My 2cents, we should not be relying on environment variables anywhere. Its just a bad idea and something i mentioned we should try and avoid previously.

Reply:
> My 2cents, we should not be relying on environment variables anywhere. Its just a bad idea and something i mentioned we should try and avoid previously.

That said, we are unlikely to change how the quick start works at this stage in the game, so what is being proposed makes sense.

Reply:
Yeah, I'd agree with that principle too - but, like you allude to, I think we'd have to unwind most of our quickstart to do so. bash isn't great for passing args/params around generally given its nature

Have refactored configureController a bit to use a switch instead